_drafts/debug-vs-release-builds.md

```diff
@@ -0,0 +1,202 @@
+---
+title: "Pocket article: Debug vs. Release Builds Considered Harmful"
+# title: "Pocket article: Debug vs. Release Builds and why you shouldn't separate them"
+description: "Why having separate Debug/Release builds should be avoided"
+author: noah
+image:
+---
+
+<!-- excerpt start -->
+
+Although having separate debug and release builds may appear to be a reasonable
+practice in embedded software development, it can actually result in more issues
+than it solves. Maintaining two different builds is burdensome, and it can lead
+to inconsistencies between the two, which can cause confusion and frustration
+during development.
+
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+## Debug vs. Release Builds for Embedded Applications
+
+Many embedded development IDEs provide separate "Debug" and "Release" build
+configurations out of the box. Generally the differences between those can be
+summarized as:
+
+| Build config | Debug info enabled?          | Compiler optimization?      | `NDEBUG` preprocessor definition?          |
+| ------------ | ---------------------------- | --------------------------- | ------------------------------------------ |
+| **Debug**    | ✅ enabled (`-g` or similar) | low (`-O0` or `-Og`)        | `DEBUG` is set, or `#define NDEBUG 0`, etc |
+| **Release**  | ❌ disabled                  | high (`-O3` or `-Os`/`-Oz`) | `NDEBUG` is set                            |
+
+Let's go one-by-one through these items!
+
+## Debug info
+
+Debug info ("debugging information") is metadata that's included in the output
+symbol file (typically `project.elf`/`project.out`/`project.axf`, but almost
+always it's an ELF file).
```

Comment:
Might be good to say ELF is used regardless of the file's extension? I guess that's not technically correct with .axf I think?

Reply:
axf is elf too! wish these ide's had used .elf to start but too late now i suppose